evap/results/views.py

```diff
@@ -62,9 +62,12 @@ def course_detail(request, semester_id, course_id):
     if not request.user.is_staff:
         # remove TextResults if user is neither the evaluated person (or a delegate) nor responsible for the course (or a delegate)
         for section in sections:
-            if not user_can_see_textresults(request.user, course, section):
-                for i, result in list(enumerate(section.results))[::-1]:
-                    if isinstance(result, TextResult):
+            for i, result in list(enumerate(section.results))[::-1]:
+                if isinstance(result, TextResult):
+                    for j, answer in list(enumerate(result.answers))[::-1]:
+                        if not user_can_see_text_answer(request.user, course, answer):
+                            del result.answers[j]
+                    if not result.answers:
```

Comment:
unfortunately named tuples are immutable and can't just be updated. i'll have a look into this later.